fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java

```diff
@@ -309,16 +310,11 @@ private Group mergeGroup(Group source, Group destination) {
         if (source.equals(destination)) {
             return source;
         }
-        List<GroupExpression> needReplaceChild = Lists.newArrayList();
-        groupExpressions.values().forEach(groupExpression -> {
-            if (groupExpression.children().contains(source)) {
-                if (groupExpression.getOwnerGroup().equals(destination)) {
-                    // cycle, we should not merge
-                    return;
-                }
-                needReplaceChild.add(groupExpression);
-            }
-        });
+        List<GroupExpression> needReplaceChild = groupExpressions.values().stream()
+                .filter(groupExpression -> groupExpression.children().contains(source))
+                // cycle, we should not merge
+                .filter(groupExpression -> !groupExpression.getOwnerGroup().equals(destination))
+                .collect(Collectors.toList());
```

Comment:
the new statement do not equal with the old one. Since old one will return immediately if any groupExpression's owner group is not destination